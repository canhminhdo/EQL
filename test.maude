in QEL.maude

--- test for mSize
red mSize(|1><1|) == 1 .
red mSize(|1><1| (x) |0><1|) == 2 .
red mSize(|1><1| (x) |0><1| (x) |1><0|) == 3 .
red mSize(|1><1| (x) |0><1| (x) |1><0| (x) |1><1|) == 4 .
--- test for vSize
red vSize(|1>) == 1 .
red vSize(|0> (x) |1>) == 2 .
red vSize(|0> (x) |0> (x) |1>) == 3 .
red vSize(|0> (x) |0> (x) |1> (x) |1>) == 4 .