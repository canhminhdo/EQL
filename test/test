#!/bin/sh

DIR=test
cd ../ && maude < test.maude -no-banner -no-advise | sed -r '/^(rewrites:|Solution|states:)/d' > $DIR/test.out
diff $DIR/test.expected $DIR/test.out
rm $DIR/test.out