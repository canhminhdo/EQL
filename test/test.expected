==========================================
fmod SCALAR
==========================================
fmod MATRIX
==========================================
fmod TENSOR
==========================================
fmod Q-ACTION
==========================================
fmod CHOICES
==========================================
fmod TELEPORT
Reading in file: "debug.maude"
Done reading in file: "debug.maude"
==========================================
reduce in TELEPORT : mSize(|1><1|) == 1 .
result Bool: true
==========================================
reduce in TELEPORT : mSize(|1><1|(x)|0><1|) == 2 .
result Bool: true
==========================================
reduce in TELEPORT : mSize(|1><1|(x)|0><1|(x)|1><0|) == 3 .
result Bool: true
==========================================
reduce in TELEPORT : mSize(|1><1|(x)|0><1|(x)|1><0|(x)|1><1|) == 4 .
result Bool: true
==========================================
reduce in TELEPORT : vSize(|1>) == 1 .
result Bool: true
==========================================
reduce in TELEPORT : vSize(|0>(x)|1>) == 2 .
result Bool: true
==========================================
reduce in TELEPORT : vSize(|0>(x)|0>(x)|1>) == 3 .
result Bool: true
==========================================
reduce in TELEPORT : vSize(|0>(x)|0>(x)|1>(x)|1>) == 4 .
result Bool: true
==========================================
reduce in TELEPORT : |0><0|(x)|0><1| x |0> == errV .
result Bool: true
==========================================
reduce in TELEPORT : |0><0|(x)|0><1| x |0><1| x |0>(x)|1> == errV .
result Bool: true
==========================================
reduce in TELEPORT : |0><1| x |0>(x)|1> == errV .
result Bool: true
==========================================
reduce in TELEPORT : |0><0|(x)|0><1| x |0> == errV .
result Bool: true
==========================================
reduce in TELEPORT : (|0><0| + |0><1|)(x)|1><1| x |0> == errV .
result Bool: true
Bye.
